File: algorithm/Python3_version/leetcode554.brick-wall.py
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        pos = 0
        n = len(wall)
        minbrick = float("inf")
        dict = {}
        for i in range(n):
            pos = 0
            for j in range(len(wall[i])-1):
                pos += wall[i][j]
                if pos not in dict:
                    dict[pos] = 1
                else:
                    dict[pos] += 1
        for p in dict:
            minbrick = min(minbrick, n-dict[p])
        if minbrick==float("inf"):
            return n
        return minbrick

File: algorithm/leetcode554.brick-wall.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
    	int pos = 0, n = wall.size(), minbrick = INT_MAX;
    	unordered_map<int,int> m;//pos，计数
    	for(int i = 0, j; i < n; i++) 
    	{
    		pos = 0;
    		for(j = 0; j < wall[i].size()-1; j++)
    		{
    			pos += wall[i][j];
    			m[pos]++;
    		}
    	}
    	for(auto it = m.begin(); it != m.end(); ++it)
    		minbrick = min(minbrick, n-it->second);
    	return minbrick==INT_MAX ? n : minbrick;
    }
};